org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/JavaClientConnection.java

```diff
@@ -140,6 +150,38 @@ public boolean applyWorkspaceEdit(WorkspaceEdit edit){
 		return response.getApplied().booleanValue();
 	}
 
+	/**
+	 * @see {@link org.eclipse.lsp4j.services.LanguageClient#unregisterCapability(RegistrationParams)}
+	 */
+	public void unregisterCapability(UnregistrationParams params) {
+		client.unregisterCapability(params).join();

```

Comment:
don't join, that makes the server unresponsive, waiting for a response that never comes.